src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -838,16 +851,40 @@ final void ensureWindowUpdated(DataFrame df) {
         }
     }
 
+    private record PushContinuationState(HeaderDecoder pushContDecoder, PushPromiseFrame pushContFrame) {}
+    private volatile PushContinuationState pcs;
```

Comment:
I'd prefer to have a longer name than `pcs` - `pushContinuationState` would make it less mysterious at the place where it's used.  Also it could be good to move these declarations (linew 854 and 855) higher in the file, where other instance variables are declared.

Reply:
Changed to pushContinuationState, its a long name but definitely clearer